File: zlcore/src/main/java/com/zaitunlabs/zlcore/utils/Hawk.java
package com.zaitunlabs.zlcore.utils;

import android.content.Context;
import android.text.TextUtils;

/**
 * Created by ahmad s on 3/2/2016.
 */
public class Hawk{
    private static Context appContext;
    public static void remove(String field){
        Prefs.with(appContext).remove(field);
    }

    public static String get(String key, String defaultValue){
        return Prefs.with(appContext).getString(key, defaultValue);
    }
    public static int get(String key, int defaultValue){
        return Prefs.with(appContext).getInt(key, defaultValue);
    }
    public static boolean get(String key, boolean defaultValue){
        return Prefs.with(appContext).getBoolean(key, defaultValue);
    }

    public static void put(String key, String value){
        if(TextUtils.isEmpty(key))return;
        Prefs.with(appContext).save(key, value);
    }

    public static void put(String key, long value){
        if(TextUtils.isEmpty(key))return;
        Prefs.with(appContext).save(key, value);
    }

    public static long get(String key, long defaultValue){
        return Prefs.with(appContext).getLong(key, defaultValue);
    }

    public static void put(String key, int value){
        if(TextUtils.isEmpty(key))return;
        Prefs.with(appContext).save(key, value);
    }

    public static void put(String key, boolean value){
        if(TextUtils.isEmpty(key))return;
        Prefs.with(appContext).save(key, value);
    }

    public static void init(Context context){
        appContext = context.getApplicationContext();
    }
}

File: zlcore/src/main/java/com/zaitunlabs/zlcore/utils/Hawk.kt
package com.zaitunlabs.zlcore.utils

import android.content.Context
import android.text.TextUtils

/**
 * Created by ahmad s on 3/2/2016.
 */
object Hawk {
    private var appContext: Context? = null
    fun remove(field: String) {
        Prefs.with(appContext).remove(field)
    }

    operator fun get(key: String, defaultValue: String): String? {
        return Prefs.with(appContext).getString(key, defaultValue)
    }

    operator fun get(key: String, defaultValue: Int): Int {
        return Prefs.with(appContext).getInt(key, defaultValue)
    }

    operator fun get(key: String, defaultValue: Boolean): Boolean {
        return Prefs.with(appContext).getBoolean(key, defaultValue)
    }

    fun put(key: String, value: String) {
        if (TextUtils.isEmpty(key)) return
        Prefs.with(appContext).save(key, value)
    }

    fun put(key: String, value: Long) {
        if (TextUtils.isEmpty(key)) return
        Prefs.with(appContext).save(key, value)
    }

    operator fun get(key: String, defaultValue: Long): Long {
        return Prefs.with(appContext).getLong(key, defaultValue)
    }

    fun put(key: String, value: Int) {
        if (TextUtils.isEmpty(key)) return
        Prefs.with(appContext).save(key, value)
    }

    fun put(key: String, value: Boolean) {
        if (TextUtils.isEmpty(key)) return
        Prefs.with(appContext).save(key, value)
    }

    fun init(context: Context) {
        appContext = context.applicationContext
    }
}